app/components/search-facet-daterange/component.js

```diff
@@ -76,16 +89,21 @@ export default Ember.Component.extend({
         }
     }),
 
-    buildQueryObject(start, end) {
-        let key = this.get('key');
-        return dateRangeFilter(key, start, end);
+    buildQueryObject(start, end, dateType) {
+        return dateRangeFilter(dateType, start, end);
     },
 
-    updateFilter(start, end) {
-        let key = this.get('key');
-        let value = start && end ? { start: moment(start).format(DATE_FORMAT), end: moment(end).format(DATE_FORMAT) } : { start: '', end: '' };
+    updateFilter(start, end, dateType) {
+        const options = this.get('filterOptions');
+        for (const option in options) {
+            if (options[option].filterBy === dateType) {
+                this.set('filterDisplay', options[option].display);
+                break;
+            }
+        }
+        let value = start && end ? { start: moment(start).format(DATE_FORMAT), end: moment(end).format(DATE_FORMAT), date: dateType } : { start: '', end: '', date: dateType };
```

Comment:
Could you add `()`s around `start && end`? `&&` does have a higher precedence than `? ... : ...`, so there's nothing wrong with it.

My rule of thumb for this is: if you have to stop and think about it (and there aren't ~3 other sets of parens), add parents.

If you have strong feelings about it, feel free to keep it as is.